typescript/widgets/src/walletIntegrations/ethereum.ts

```diff
@@ -159,6 +160,38 @@ export function useEthereumTransactionFns(
   return { sendTransaction: onSendTx, switchNetwork: onSwitchNetwork };
 }
 
+export function useEthereumSignTypedData(multiProvider: MultiProtocolProvider) {
```

Comment:
This hook was added to allow users to sign an `EIP-712` message. It’s data-agnostic, so it can be reused if needed.